crates/oxc_semantic/src/scope.rs

```diff
@@ -38,7 +43,30 @@ impl ScopeTree {
         std::iter::successors(Some(scope_id), |scope_id| self.parent_ids[*scope_id])
     }
 
-    pub fn descendants(&self) -> impl Iterator<Item = ScopeId> + '_ {
+    pub fn descendants(&self, scope_id: ScopeId) -> impl Iterator<Item = ScopeId> + '_ {
+        let mut list = vec![];
+
+        // Has to be a `fn` and pass arguments because we can't
+        // have recursive closures
+        fn add_to_list(
+            parent_id: &ScopeId,
+            child_ids: &IndexMap<ScopeId, Vec<ScopeId>>,
+            items: &mut Vec<ScopeId>,
+        ) {
+            if let Some(children) = child_ids.get(parent_id) {
+                for child_id in children {
+                    items.push(*child_id);
```

Comment:
Would it make a little bit faster with a `with_capacity` before the loop?

Reply:
Don't have a length before hand.